### PR TITLE
feat: scheduled deliveries filter state

### PR DESCRIPTION
**Pull request by IrakliJani**
Related to: https://github.com/lightdash/lightdash/issues/4425

### Description:

adds state management to scheduled delivery filters

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx [hunk 1]
```diff
@@ -24,19 +26,29 @@ import {
 import { useProject } from '../../../hooks/useProject';
 import { useDashboardContext } from '../../../providers/DashboardProvider';
 
+type SchedulerFilterRule = DashboardFilterRule & {
+    tileTargets: undefined;
```
Comment (joaoviana) [inline packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx:30]: do we want it `undefined` or optional? 
maybe... `optional`? so we don't need to add this line. What do you think?

**Review by joaoviana — APPROVED** [review]
one Q on types 👍

File: packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx:30]: @joaoviana good question! ok so I was thinking about this a lot. whenever you update filter it returns new `DashboardFilterRule` and it includes `tileTargets`.

typescript does not complain whenever you have additional properties in the object so it is really easy to forget removing that property and thought to make it explicit - `SchedulerFilterRule` should  contain `tileTargets` equal to undefined.

let me know if this is useless and we should take care of it on the BE side?

Comment (joaoviana) [reply packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx:33]: ahhhhh! I see!! Amazing, let's keep your change! 👍

**Pull request merged**